source/extensions/resource_monitors/cpu_utilization/cpu_utilization_monitor.cc

```diff
@@ -27,13 +27,37 @@ constexpr double DAMPENING_ALPHA = 0.05;
 
 CpuUtilizationMonitor::CpuUtilizationMonitor(
     const envoy::extensions::resource_monitors::cpu_utilization::v3::
-        CpuUtilizationConfig& /*config*/,
-    std::unique_ptr<CpuStatsReader> cpu_stats_reader)
-    : cpu_stats_reader_(std::move(cpu_stats_reader)) {
+        CpuUtilizationConfig& config,
+    std::unique_ptr<CpuStatsReader> cpu_stats_reader,TimeSource &time_source)
+    : cpu_stats_reader_(std::move(cpu_stats_reader)),time_source_(time_source),last_update_time_(time_source.monotonicTime()),mode_(config.mode()) {
   previous_cpu_times_ = cpu_stats_reader_->getCpuTimes();
 }
 
+CpuUtilizationMonitor::CpuUtilizationMonitor(
+  const envoy::extensions::resource_monitors::cpu_utilization::v3::
+        CpuUtilizationConfig& config,
+  std::unique_ptr<CgroupStatsReader> cgroup_stats_reader,TimeSource &time_source)
+  : cgroup_stats_reader_(std::move(cgroup_stats_reader)),time_source_(time_source),last_update_time_(time_source.monotonicTime()),mode_(config.mode()) {
+    previous_cgroup_stats_ = cgroup_stats_reader_->getCgroupStats();
+  } 
+
 void CpuUtilizationMonitor::updateResourceUsage(Server::ResourceUpdateCallbacks& callbacks) {
+  switch (mode_)
+  {
+  case envoy::extensions::resource_monitors::cpu_utilization::v3::CpuUtilizationConfig::HOST:
+    computeHostCpuUsage(callbacks);
+    break;
+  
+  case envoy::extensions::resource_monitors::cpu_utilization::v3::CpuUtilizationConfig::CONTAINER:
+    computeContainerCpuUsage(callbacks);
+    break;
+
+  default:
```

Comment:
prefer no default here since we control the enum as this will force exhaustive switch conditions